packages/markdown/src/lib/MarkdownPlugin.ts

```diff
@@ -6,26 +8,68 @@ import {
   isUrl,
 } from '@udecode/plate';
 
-import { deserializeMd } from './deserializer/utils';
-import {
-  type RemarkElementRules,
-  type RemarkTextRules,
-  remarkDefaultElementRules,
-  remarkDefaultTextRules,
-} from './remark-slate';
+import type { TNodes } from './nodesRule';
+// import type { deserializeMd } from './deserializer/deserializeMd';
+import type { plateTypes } from './utils/mapTypeUtils';
+
+import { deserializeMd } from './deserializer';
 import { serializeMd } from './serializer';
 
-// export type MarkdownDeserializer = {
-//   elementRules?: Partial<Record<MdastElementType, RemarkElementRule>>;
-//   textRules?: Partial<Record<MdastTextType, RemarkTextRule>>;
-// } & Deserializer;
+export type AllowNodeConfig = {
+  /** Custom filter function for nodes during deserialization */
+  deserialize?: (node: any) => boolean;
+  /** Custom filter function for nodes during serialization */
+  serialize?: (node: any) => boolean;
+};
 
 export type MarkdownConfig = PluginConfig<
   'markdown',
   {
-    /** Override element rules. */
-    elementRules?: RemarkElementRules;
-    indentList?: boolean;
+    /**
+     * Configuration for allowed node types. Cannot be combined with
+     * disallowedNodes.
+     */
+    allowedNodes: NodesConfig;
+    /**
+     * Configuration for disallowed node types. Cannot be combined with
+     * allowedNodes.
+     *
+     * @default null
+     */
+    disallowedNodes: NodesConfig;
+    /**
+     * Array of remark plugins to extend Markdown parsing and serialization
+     * functionality. For example, you can add remark-gfm to support GFM syntax,
+     * remark-math to support mathematical formulas, etc. These plugins will be
+     * used during the parsing and generation of Markdown text.
+     *
+     * @default undefined
+     */
+    remarkPlugins: Plugin[];
+    /**
+     * Custom filter function for nodes during deserialization and
+     * serialization.
+     *
+     * @default null
+     */
+    allowNode?: AllowNodeConfig;
```

Comment:
Missing the default below. We can't use optional, otherwise the jotai store will not have the initial atom.